runtime/lua/vim/fs.lua

```diff
@@ -74,7 +74,11 @@ end
 
 ---@private
 local function join_paths(...)
-  return table.concat({ ... }, '/')
+  local path_elements = vim.tbl_map(function(element)
+    return element:gsub('/$', '')
+  end, { ... })
+
+  return table.concat(path_elements, '/')
```

Comment:
How about just this? It will remove all empty segments from the path.
```suggestion
  return (table.concat({ ... }, '/'):gsub('//+', '/'))
```

Reply:
I thought about that, but in general I try to avoid masking errors in input. E.g. it'll just magically work for:

```lua
vim.inspect(vim.fs.find('something', {
  path = 'broken////////////////input',
  upward = true,
  type = 'directory',
}))
```

But if you prefer the simplicity, I don't feel too strongly about it.

Reply:
This path is still valid though, at least on Linux.

Reply:
I think the main concern is performance. This function could be used in hot loops.